mithril-client/src/cardano_database_client/download_unpack.rs

```diff
@@ -0,0 +1,1204 @@
+            let immutable_file_numbers_downloaded_chunk = self
+                .batch_download_unpack_immutable_files(
+                    file_downloader.clone(),
+                    file_downloader_uris_chunk,
+                    compression_algorithm,
+                    immutable_files_target_dir,
+                    download_id,
+                    file_size,
+                )
+                .await?;
+            immutable_file_numbers_downloaded.extend(immutable_file_numbers_downloaded_chunk);
+        }
+
+        Ok(immutable_file_numbers_downloaded)
+    }
+
+    /// Download and unpack the ancillary files.
+    pub(crate) async fn download_unpack_ancillary_file(
+        &self,
+        locations: &[AncillaryLocation],
+        compression_algorithm: &CompressionAlgorithm,
+        ancillary_file_target_dir: &Path,
+        download_id: &str,
+    ) -> MithrilResult<()> {
+        let mut locations_sorted = locations.to_owned();
+        locations_sorted.sort();
+        for location in locations_sorted {
+            // The size will be completed with the uncompressed file size when available in the location
+            // (see https://github.com/input-output-hk/mithril/issues/2291)
+            let file_size = 0;
+            self.feedback_sender
+                .send_event(MithrilEvent::CardanoDatabase(
+                    MithrilEventCardanoDatabase::AncillaryDownloadStarted {
+                        download_id: download_id.to_string(),
+                        size: file_size,
+                    },
+                ))
+                .await;
+            let file_downloader = match &location {
+                AncillaryLocation::CloudStorage { .. } => self.http_file_downloader.clone(),
+            };
+            let file_downloader_uri = location.into();
+            let downloaded = file_downloader
+                .download_unpack(
+                    &file_downloader_uri,
+                    ancillary_file_target_dir,
+                    Some(compression_algorithm.to_owned()),
+                    DownloadEvent::Ancillary {
+                        download_id: download_id.to_string(),
+                    },
+                )
+                .await;
+            match downloaded {
+                Ok(_) => {
+                    self.feedback_sender
+                        .send_event(MithrilEvent::CardanoDatabase(
+                            MithrilEventCardanoDatabase::AncillaryDownloadCompleted {
+                                download_id: download_id.to_string(),
+                            },
+                        ))
+                        .await;
+                    return Ok(());
+                }
+                Err(e) => {
+                    slog::error!(
+                        self.logger,
+                        "Failed downloading and unpacking ancillaries for location {file_downloader_uri:?}"; "error" => e.to_string()
```

Comment:
Should we debug the error to show the stack trace ? Else only the main message will be shown.
```suggestion
                        "Failed downloading and unpacking ancillaries for location {file_downloader_uri:?}"; "error" => ?e
```